flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/DistinctQueryOperation.java

```diff
@@ -45,6 +45,13 @@ public String asSummaryString() {
                 "Distinct", Collections.emptyMap(), getChildren(), Operation::asSummaryString);
     }
 
+    @Override
+    public String asSerializableString() {
+        return String.format(
+                "SELECT DISTINCT * FROM (%s\n)",
```

Comment:
Can you double check whether we need to expand all stars in operations? Otherwise this could interfere with virtual metadata columns given the expansion behavior is configured. Check `ColumnExpansionTest#testExcludeDefaultVirtualMetadataColumns`.

Reply:
I changed all `*` to an explicit list.